.github/workflows/nightly.yml

```diff
@@ -280,7 +280,7 @@ jobs:
       fail-fast: true
       matrix:
         build:
-          - { os: macos-11, xcode: 13.2.1, deployment: 11.3 }
+          - { os: macos-12, xcode: 14.2, deployment: 12.0 }
```

Comment:
macos-12 runner is on macos 12.7 so that should be fine for Xcode 14.2.
The deployment target can still address macOS 12.0

Reply:
And xcode 13.2, despite requiring macos 11.3 to run, could have easily targeted macos 11.0 / earlier,
yet we've pretty intentionally decided to avoid dealing with backdeploy horror,
by targeting the same macos version that is required for the xcode to run.

Reply:
To reword: yes, in general this change is very much good, and i was going to do that
once macos-14 becomes available on github actions and thus macos-11 becomes EOL.

But, it is really non-obvious to me why must the `deployment` target *NOT* be `12.5`.
Unless there are very specific reasons not to, we should go with `12.5`, not `12.0`.

Reply:
> > Are you sure about `deployment` target? Shouldn't it be `12.5`?
> 
> I'm pretty sure the deployment target (whether defined as a runner environment variable or even hardcoded somewhere, doesn't matter) is _not used_ in CI and nightly darktable builds. Therefore at the moment it doesn't matter what is written there.
> 
> What to do with it (discard the deployment target as unnecessary in practice or vice versa implement management of deployment targets in builds) is the topic of a separate conversation and I was just going to raise an issue about it after fixing the nightly build.

[Citation needed.](https://github.com/darktable-org/darktable/blob/aa2faf791456b1b125c62b298b6967f1a89016de/.github/workflows/nightly.yml#L296C7-L296C31)

Reply:
>  is not used in CI and nightly darktable builds

It sets the env variable `MACOSX_DEPLOYMENT_TARGET` which gets passed to the compiler and is finally embedded in the binary (see the value for `minos`):

```
vtool -show-build darktable.app/Contents/MacOS/darktable        
darktable.app/Contents/MacOS/darktable:
Load command 10
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform MACOS
    minos 11.3
      sdk 12.1
   ntools 1
     tool LD
  version 711.0
```

Reply:
Finally, I am at home and can work normally. Answering from a mobile phone at stops in front of traffic lights is somehow not very convenient. :)

> It sets the env variable `MACOSX_DEPLOYMENT_TARGET` which gets passed to the compiler

@zisoft Thanks for the hint, for some reason I only checked passing arguments to the compiler. Googling revealed that Xcode just takes into account the MACOSX_DEPLOYMENT_TARGET environment variable itself. OK, I'm raising the deployment target version.